### ISSUE TITLE
Strange cell probability plot in CellBender output

### ISSUE THREAD

**Issue by huipan1973** [broadinstitute/CellBender#442]
[S1-1A_cellbender.log.txt](https://github.com/user-attachments/files/25817371/S1-1A_cellbender.log.txt)
[S1-1A_cellbender.pdf](https://github.com/user-attachments/files/25817370/S1-1A_cellbender.pdf)

Hello CellBender developers,

Thank you for developing and maintaining CellBender — it has been very useful for our scRNA-seq preprocessing.

I recently ran CellBender remove-background v0.3.2 on a 10x Genomics dataset, but the cell probability plot in the HTML report looks very unusual, and I wanted to ask whether this behavior is expected or indicative of a problem.

Command used
cellbender remove-background \
  --input raw_feature_bc_matrix.h5 \
  --output S1-1A_cellbender.h5 \
  --expected-cells 20845 \
  --total-droplets-included 62535 \
  --fpr 0.01 \
  --epochs 150 \
  --model full \
  --low-count-threshold 10 \
  --cpu-threads 20 \
  --debug
Dataset summary

From the CellBender log:

Features detected: 33,697 genes

Features used for inference: 15,917

Expected cells (user-specified): 20,845

Automatically estimated cells: ~10,456

Transition point detected: 21,002 droplets

Total droplets included: 62,535

Automatically computed priors:
expected_cells: 10456
transition_point: 21002
Fixing expected_cells at 20845
Fixing total_droplets at 62535

Full log attached.

Training behavior

The ELBO appears to converge reasonably during training.

The training loss decreases early and then fluctuates slightly but remains stable, which seems normal for this model.

Strange probability plot

However, the cell probability plot appears abnormal:

Instead of a smooth transition between cells and empty droplets, there are vertical bands of probabilities near 0 and 1.

A large fraction of droplets across the entire barcode range have high cell probabilities, even where UMI counts are extremely low.

Example plot:

S1-1A_cellbender

Expected behavior (based on the documentation) would be a smoother probability transition around the knee point.

Questions

Is this probability distribution expected for some datasets, or does it indicate a model fitting issue?

Could this behavior arise from forcing --expected-cells 20845, while CellBender estimated ~10,456 cells automatically?

Are there recommended diagnostics to confirm whether the inferred cell probabilities are reliable?

Would you recommend rerunning with:

automatic expected cells

different total-droplets-included

a different model (e.g., --model ambient)?

Environment

CellBender version: 0.3.2

Input format: 10x CellRanger v3

OS: Linux (HPC cluster)

Attachments:

HTML report

full log

Any guidance would be greatly appreciated.

Thank you!
